test/lint/lint-circular-dependencies.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020-2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check for circular dependencies
+
+import glob
+import os
+import re
+import subprocess
+import sys
+
+EXPECTED_CIRCULAR_DEPENDENCIES = (
+    "chainparamsbase -> util/system -> chainparamsbase",
+    "node/blockstorage -> validation -> node/blockstorage",
+    "index/blockfilterindex -> node/blockstorage -> validation -> index/blockfilterindex",
+    "index/base -> validation -> index/blockfilterindex -> index/base",
+    "index/coinstatsindex -> node/coinstats -> index/coinstatsindex",
+    "policy/fees -> txmempool -> policy/fees",
+    "qt/addresstablemodel -> qt/walletmodel -> qt/addresstablemodel",
+    "qt/recentrequeststablemodel -> qt/walletmodel -> qt/recentrequeststablemodel",
+    "qt/sendcoinsdialog -> qt/walletmodel -> qt/sendcoinsdialog",
+    "qt/transactiontablemodel -> qt/walletmodel -> qt/transactiontablemodel",
+    "wallet/fees -> wallet/wallet -> wallet/fees",
+    "wallet/wallet -> wallet/walletdb -> wallet/wallet",
+    "node/coinstats -> validation -> node/coinstats",
+)
+
+CODE_DIR = "src"
+
+
+def main():
+    circular_dependencies = []
+    exit_code = 0
+    os.chdir(
+        CODE_DIR
+    )  # We change dir before globbing since glob.glob's root_dir option is only available in Python 3.10
+
+    # Using glob.glob since subprocess.run's globbing won't work without shell=True
+    files = []
+    for path in ["*", "*/*", "*/*/*"]:
+        for extension in ["h", "cpp"]:
+            files.extend(glob.glob(f"{path}.{extension}"))
+
+    command = ["python3", "../contrib/devtools/circular-dependencies.py", *files]
+    dependencies_output = subprocess.run(
+        command,
+        stdout=subprocess.PIPE,
```

Comment:
any reason to ignore errors?

```suggestion
    dependencies_output = subprocess.check_output(
        command,
```

Reply:
Because `../contrib/devtools/circular-dependencies.py` does not exits with 0 if there is at least one circular dependency, resulting in `subprocess.check` raising an error